sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/SchedulerBuilder.java

```diff
@@ -246,6 +221,56 @@ public SchedulerBuilder setTemplateUrlFactory(ArtifactQueries.TemplateUrlFactory
      * @throws IllegalArgumentException if validating the provided configuration failed
      */
     public AbstractScheduler build() {
+        // If region awareness is enabled (via java bit or via env) and the cluster supports it, update the ServiceSpec
+        // to include region constraints.
+        final ServiceSpec serviceSpec;
+        if (Capabilities.getInstance().supportsDomains()) {
```

Comment:
cc @mrbrowning for sanity check on region PlacementRule init being performed here (also wrote related tests in `SchedulerBuilderTest.java`